### DISCUSSION TITLE
help configuration enphase

### DISCUSSION THREAD

**Discussion by JeremC82** [xlyric/pv-router-esp32#36] [General]
Hello, I have just loaded = 
 -firmware-esp32.bin
 -spiffs-dimmer.bin

my esp is well connected to my wifi, I can access the web page. I can't find the information regarding the configuration with an enphase envoy. Can you help me ? Thank you very much for your support

**Comment by xlyric** [comment]
Hi, 

go on the /envoy.html page 
and after 
you can configure the envoy 

server  - port  ( wrong typo ) -  model ( S or R ) - version ( 5 or 7 ) - and the token is given by your envoy ( copy/past the token given by your web explorer )

**Reply by JeremC82** [reply]
sorry , , to be sure not to make a mistake I take this carefully

https://ota.apper-solaire.org/firmware/firmware-ttgo.bin

https://ota.apper-solaire.org/firmware/spiffs-ttgo.bin

**Reply by JeremC82** [reply]
see attached the log of terminal
 enphase config : 192.168.2.10
 enphase version : 7
 enphase mode : s
 enphase : inactif
start 18b20
Dallas not connected

Why enphase mode is disable ?

**Comment by xlyric** [comment]
Hi,

I will check if there is a problem on my board.

cordially
Cyril

Le lun. 20 nov. 2023 à 17:22, MYGIT ***@***.***> a écrit :

> see attached the log of terminal
> enphase config : 192.168.2.10
> enphase version : 7
> enphase mode : s
> enphase : inactif
> start 18b20
> Dallas not connected
>
> Why enphase mode is disable ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/xlyric/pv-router-esp32/discussions/36#discussioncomment-7621407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABP6WB7Z5PET3ND255ECCJ3YFN7TVAVCNFSM6AAAAAA7RYK4H6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRRGQYDO>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Cordialement
Cyril P.

**Comment by xlyric** [comment]
I try in my way :

I configure the Envoy and reboot after :

#################  Restart reason  ###############
3
#################  Starting System  ###############
01:00:00 Start Filesystem
01:00:00 config file loaded
16:31:33-- reboot demande par l'utilisateur --
01:00:00 logs loaded
01:00:00 Wifi config
01:00:00 MQTT config loaded
01:00:00 --> Fronius not present
01:00:00 Enphase used type S
01:00:00 Dallas not connected
01:00:01 Start Wifi Network FreeLyric
01:00:03 Wifi connected
16:31:38 MQTT attempting reco
-57 dBm
16:31:38 MQTT : Reconnected
-- fin du demarrage: 16:31:39 --

and I look in serial, I see "enphase : inactif"
it's a error in the serial print.

I will correct the mistake

Le mar. 21 nov. 2023 à 15:05, Cyril Poissonnier ***@***.***>
a écrit :

> Hi,
>
> I will check if there is a problem on my board.
>
> cordially
> Cyril
>
> Le lun. 20 nov. 2023 à 17:22, MYGIT ***@***.***> a écrit :
>
>> see attached the log of terminal
>> enphase config : 192.168.2.10
>> enphase version : 7
>> enphase mode : s
>> enphase : inactif
>> start 18b20
>> Dallas not connected
>>
>> Why enphase mode is disable ?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/xlyric/pv-router-esp32/discussions/36#discussioncomment-7621407>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABP6WB7Z5PET3ND255ECCJ3YFN7TVAVCNFSM6AAAAAA7RYK4H6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRRGQYDO>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Cordialement
> Cyril P.
>


-- 
Cordialement
Cyril P.

**Comment by xlyric** [comment]
Hi,
I have made an update, you can keep the new .bin file.

Cordialy

Le mar. 21 nov. 2023 à 17:21, Cyril Poissonnier ***@***.***>
a écrit :

> I try in my way :
>
> I configure the Envoy and reboot after :
>
> #################  Restart reason  ###############
> 3
> #################  Starting System  ###############
> 01:00:00 Start Filesystem
> 01:00:00 config file loaded
> 16:31:33-- reboot demande par l'utilisateur --
> 01:00:00 logs loaded
> 01:00:00 Wifi config
> 01:00:00 MQTT config loaded
> 01:00:00 --> Fronius not present
> 01:00:00 Enphase used type S
> 01:00:00 Dallas not connected
> 01:00:01 Start Wifi Network FreeLyric
> 01:00:03 Wifi connected
> 16:31:38 MQTT attempting reco
> -57 dBm
> 16:31:38 MQTT : Reconnected
> -- fin du demarrage: 16:31:39 --
>
> and I look in serial, I see "enphase : inactif"
> it's a error in the serial print.
>
> I will correct the mistake
>
> Le mar. 21 nov. 2023 à 15:05, Cyril Poissonnier <
> ***@***.***> a écrit :
>
>> Hi,
>>
>> I will check if there is a problem on my board.
>>
>> cordially
>> Cyril
>>
>> Le lun. 20 nov. 2023 à 17:22, MYGIT ***@***.***> a écrit :
>>
>>> see attached the log of terminal
>>> enphase config : 192.168.2.10
>>> enphase version : 7
>>> enphase mode : s
>>> enphase : inactif
>>> start 18b20
>>> Dallas not connected
>>>
>>> Why enphase mode is disable ?
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/xlyric/pv-router-esp32/discussions/36#discussioncomment-7621407>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ABP6WB7Z5PET3ND255ECCJ3YFN7TVAVCNFSM6AAAAAA7RYK4H6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRRGQYDO>
>>> .
>>> You are receiving this because you commented.Message ID:
>>> ***@***.***>
>>>
>>
>>
>> --
>> Cordialement
>> Cyril P.
>>
>
>
> --
> Cordialement
> Cyril P.
>


-- 
Cordialement
Cyril P.

**Comment by JeremC82** [comment]
Je viens de voir que tu étais Français , cela facilitera nos échanges . 

voici mon resultat : 

192.168.2.10
 enphase config : 192.168.2.10
 enphase version : 7
 enphase mode : s
 enphase : actif
 enphase token : (montoken )
start 18b20
Dallas not connected

.WiFi connected
IP address:
192.168.2.89
force du signal:-78dBm
Démarrage Oled
17:46:32
Dimmer started...
Loading minuterie
start Web server
Get Dimmer temp : Error code: -1
Get Dimmer temp : Error code: -1


Je suis sur un esp32 dev module monte sur un pcb perso 
    #define zerocross  35 // for boards with CHANGEBLE input pins // ZC on board
    #define ONE_WIRE_BUS  15
    #define OUTPUTPIN1  32 // PSM on board

Les valeurs sigma remonté dans la page ne corresponde pas a la réalité d ailleurs si je configure un fausse ip pour l envoy j ai le même comportement 
est il possible dans le moniteur série d avoir un debug pour voir si la requête vers l envoy est ok ainsi que ce quel retourne ?

Merci encor pour ton support ainsi que ta patience 

Ps un petit petit bouton dans le menu config pour passer en mode envoy serait top avec une libération des champs lorsque l option est activé :)

**Comment by xlyric** [comment]
ok, 
pour la partie enphase, c'est multinet33 qui a fait cette partie. 
( surtout pour le token ) 

pour l'enphase, il se met en route si il y a une ip dans le champs du host, sinon c'est la sonde locale qui tourne. 

coté débug, il indique dans le sérial l'url qu'il appelle . 
( L217 dans enphaseFunction.h ) 

le problème est qu'en fonction de ce que sort enphase, la donnée dans le Json peu changer. 

en principe il se base sur  gDisplayValues.Fronius_totalconso = int(doc["consumption"][0]["wNow"]);

vu que je n'ai pas d'enphase, pour le début, le mieux est de voir dans le Json qu'il sort et trouver la valeur qui corresponds. 



après le fait que tu ai fait une carte perso, ça va pas changer grand chose en principe. ( au delà du fait que ça ne fait pas rentrer des sous dans l'assos ;)

**Comment by xlyric** [comment]
avec le json que j'avais, sur une version 5 il fait le boulot. à 1ere vu, je constate pas de pb d'affichage coté interface web 
les valeurs sont cohérentes

donc si t'as le Json pour la version 7 
( au passage on m'a remonté que c'est un peu la M le token envoy, il est valable 24H sinon fait faire une demande chez eux

**Comment by JeremC82** [comment]
effectivement , apres il existe un moyen de renouveler le token automatiquement via une requette http . 
du coup avez vous une mise a jour du token automatique ? 

Voici par exemple le json fournis pas mon envoy 
`{
    "production": [
        {
            "type": "inverters",
            "activeCount": 8,
            "readingTime": 1700771813,
            "wNow": 0,
            "whLifetime": 426537
        },
        {
            "type": "eim",
            "activeCount": 1,
            "measurementType": "production",
            "readingTime": 1700771840,
            "wNow": -9.827,
            "whLifetime": 6378491.114,
            "varhLeadLifetime": 1.745,
            "varhLagLifetime": 1892030.675,
            "vahLifetime": 7749542.514,
            "rmsCurrent": 0.633,
            "rmsVoltage": 231.872,
            "reactPwr": 134.448,
            "apprntPwr": 147.176,
            "pwrFactor": -0.1,
            "whToday": 8769.114,
            "whLastSevenDays": 36203.114,
            "vahToday": 10984.514,
            "varhLeadToday": 0.745,
            "varhLagToday": 2886.675
        }
    ],
    "consumption": [
        {
            "type": "eim",
            "activeCount": 1,
            "measurementType": "total-consumption",
            "readingTime": 1700771840,
            "wNow": 978.574,
            "whLifetime": 11353304.202,
            "varhLeadLifetime": 5237301.67,
            "varhLagLifetime": -1813551.807,
            "vahLifetime": 12093317.342,
            "rmsCurrent": 5.7,
            "rmsVoltage": 232.107,
            "reactPwr": -307.398,
            "apprntPwr": 1322.979,
            "pwrFactor": 0.74,
            "whToday": 22015.202,
            "whLastSevenDays": 119.202,
            "vahToday": 21966.342,
            "varhLeadToday": 7730.67,
            "varhLagToday": 0
        },
        {
            "type": "eim",
            "activeCount": 1,
            "measurementType": "net-consumption",
            "readingTime": 1700771840,
            "wNow": 988.401,
            "whLifetime": 4975565.706,
            "varhLeadLifetime": 5237303.415,
            "varhLagLifetime": 78478.868,
            "vahLifetime": 12093317.342,
            "rmsCurrent": 5.066,
            "rmsVoltage": 232.107,
            "reactPwr": -441.846,
            "apprntPwr": 1175.96,
            "pwrFactor": 0.84,
            "whToday": 0,
            "whLastSevenDays": 0,
            "vahToday": 0,
            "varhLeadToday": 0,
            "varhLagToday": 0
        }
    ],
    "storage": [
        {
            "type": "acb",
            "activeCount": 0,
            "readingTime": 0,
            "wNow": 0,
            "whNow": 0,
            "state": "idle"
        }
    ]
}`

**Comment by xlyric** [comment]
ce que j'en vois ça serait
"consumption": [ { "wNow": 978.574
qu'il faudrait capturer ?

**Reply by JeremC82** [reply]
bonjour oui exactement :)

**Reply by multinet33** [reply]
Hello

sauf erreur de ma part c'est bien ce qui est codé : 
```
    if (String(configmodule.envoy) == "R") {
      gDisplayValues.Fronius_prod = int(doc["wattsNow"]);
      gDisplayValues.Fronius_conso = int(doc["wattHoursToday"]);
    } else {

      if ((gDisplayValues.Fronius_prod = int(doc["production"][1]["wNow"])) < 0) {
        // During night, Envoy measure a negative production
        gDisplayValues.Fronius_prod = 0;
      }
      gDisplayValues.Fronius_totalconso = int(doc["consumption"][0]["wNow"]);
      gDisplayValues.Fronius_conso = gDisplayValues.Fronius_totalconso - gDisplayValues.Fronius_prod;
    }
```

**Reply by multinet33** [reply]
chez moi tout est ok sur envoy v7 s metered (type S et version 7)

**Comment by xlyric** [comment]
c'est ce que j'ai vu aussi,
du coup, c'est quoi le problème ?

**Comment by JeremC82** [comment]
Du coup si j ajoute une fausse ip dans la conf enphase (mauvaise configuration volontaire pour un test) j ai quand meme des valeur sigma qui bouge hors cela n est pas possiblie puisqu il n y a pas de communication .Sauf si je me trompe sur le fonctionnement

**Comment by xlyric** [comment]
je saisis pas trop.
tu peux me faire une capture d' écran ?
comme autant sur le TTGO je sais ce qu'il se passe et la valeur de prod et
conso envoy sont affiché en bas sur l'afficheur.
autant sur un truc perso, je ne sais pas ce que tu as fait.

après tu peux poser tes questions sur
https://forum-photovoltaique.fr/viewtopic.php?f=110&t=41777&p=730963#p730963
ils sauront surement te répondre comme ils sont plusieurs à l'utiliser.

**Reply by JeremC82** [reply]
Bonjour , désolé pour ma réponse tardive . Je vais recevoir un ttgo pour faire les test normalement demain , car j utilise un oled i2c dans mon autre projet incompatible avec le code actuelle . Je reviens au plus vite avec des valeurs sure :)

**Reply by xlyric** [reply]
techniquement avec un oled, il faut juste redéfinir ce que l'on affiche. quand j'ai commencé le projet il y a 4 ans, ça tournait sur un ESP8266 avec un écran oled. 
[https://pvrouteur.apper-solaire.org/uploads/images/gallery/2022-03/scaled-1680-/image-1648138727532.png](url)
